providers/os/resources/services/systemd.go

```diff
@@ -32,49 +32,104 @@ func ResolveSystemdServiceManager(conn shared.Connection) OSServiceManager {
 	return &SystemDServiceManager{conn: conn}
 }
 
-// a line may be prefixed with nothing, whitespace or a dot
+// Newer linux systems use systemd as service manager
+type SystemDServiceManager struct {
+	conn shared.Connection
+}
+
+func (s *SystemDServiceManager) Name() string {
+	return "systemd Service Manager"
+}
+
 func ParseServiceSystemDUnitFiles(input io.Reader) ([]*Service, error) {
 	var services []*Service
 	content, err := io.ReadAll(input)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error executing systemctl list-unit-files: %v", err)
 	}
 
-	m := SYSTEMD_LIST_UNITS_REGEX.FindAllStringSubmatch(string(content), -1)
-	for i := range m {
-		name := m[i][1]
-		name = strings.Replace(name, ".service", "", 1)
+	lines := strings.Split(string(content), "\n")
+	if len(lines) < 2 {
+		return nil, fmt.Errorf("unexpected output from systemctl list-unit-files")
+	}
 
-		s := &Service{
+	for _, line := range lines[1 : len(lines)-1] {
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			continue
+		}
+
+		name := strings.TrimSuffix(fields[0], ".service")
+
+		service := &Service{
 			Name:      name,
-			Installed: m[i][2] == "loaded",
-			Running:   m[i][3] == "active",
-			// TODO: we may need to revist the enabled state
-			Enabled:     m[i][2] == "loaded",
-			Masked:      m[i][2] == "masked",
-			Description: m[i][5],
-			Type:        "systemd",
+			Enabled:   fields[1] == "enabled",
+			Installed: fields[1] == "enabled",
```

Comment:
I'd like to propose a differentiation between _Installed_ and _Enabled_:
If you found the service through a call to systemd, it is installed. So you could probably go with `Installed: true`.
Enabled can stay the same, as it will tell the user if the service will start at boot.